cmd/clusterctl/internal/util/objs.go

```diff
@@ -105,42 +119,93 @@ func FixImages(objs []unstructured.Unstructured, alterImageFunc func(image strin
 	// look for resources of kind Deployment and alter the image
 	for i := range objs {
 		o := &objs[i]
-		if o.GetKind() != deploymentKind {
-			continue
+		err := fixDeploymentImage(o, alterImageFunc)
+		if err != nil {
+			return nil, err
 		}
-
-		// Convert Unstructured into a typed object
-		d := &appsv1.Deployment{}
-		if err := scheme.Scheme.Convert(o, d, nil); err != nil {
+		err = fixDaemonSetImage(o, alterImageFunc)
+		if err != nil {
 			return nil, err
 		}
+		objs[i] = *o
```

Comment:
I'm not sure how to properly abstract deamonset and image here. This code works but requires some code dupliation. @fabriziopandini what do you think?

Reply:
What about extracting the bits  that are modifying Containers/InitContainers? Both of them are working on []Containers and they are the same for DaemonSet & Deployments (that's means that they are repeated 4 times now...)

Reply:
What do you think now?